src/librustc_back/target/mod.rs

```diff
@@ -62,6 +62,47 @@ mod netbsd_base;
 mod windows_base;
 mod windows_msvc_base;
 
+/// List of supported targets
+pub const TARGETS: &'static [&'static str] = &[
+    "aarch64-apple-ios",
+    "aarch64-linux-android",
+    "aarch64-unknown-linux-gnu",
+    "arm-linux-androideabi",
+    "arm-unknown-linux-gnueabi",
+    "arm-unknown-linux-gnueabihf",
+    "armv7-apple-ios",
+    "armv7-unknown-linux-gnueabihf",
+    "armv7s-apple-ios",
+    "i386-apple-ios",
+    "i686-apple-darwin",
+    "i686-linux-android",
+    "i686-pc-windows-gnu",
+    "i686-pc-windows-msvc",
+    "i686-unknown-dragonfly",
+    "i686-unknown-freebsd",
+    "i686-unknown-linux-gnu",
+    "le32-unknown-nacl",
+    "mips-unknown-linux-gnu",
+    "mips-unknown-linux-musl",
+    "mipsel-unknown-linux-gnu",
+    "mipsel-unknown-linux-musl",
+    "powerpc-unknown-linux-gnu",
+    "powerpc64-unknown-linux-gnu",
+    "powerpc64le-unknown-linux-gnu",
+    "x86_64-apple-darwin",
+    "x86_64-apple-ios",
+    "x86_64-pc-windows-gnu",
+    "x86_64-pc-windows-msvc",
+    "x86_64-rumprun-netbsd",
+    "x86_64-unknown-bitrig",
+    "x86_64-unknown-dragonfly",
+    "x86_64-unknown-freebsd",
+    "x86_64-unknown-linux-gnu",
+    "x86_64-unknown-linux-musl",
+    "x86_64-unknown-netbsd",
+    "x86_64-unknown-openbsd",
+];
```

Comment:
> Would it be possible to not duplicate this list with the one below? Could the macro somehow use something like stringify! to build a vector, or could an iterator macro be used?

As @nodakai said, the mixture of hyphens and underscores makes it tricky.

> I think "x86_64-w64-mingw32" is an alias for "x86_64-pc-windows-gnu" and should be included in the list as a valid target.

That's what the logic of this function seems to indicate. But, `rustc --target=x86_64-w64-mingw32 ...` returns `error: Error loading target specification: Could not find specification for target "x86_64-w64-mingw32"`.